src/PublicProposalsPage/PublicProposalsProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useMemo, useCallback, useState, useEffect } from "react";
+import unionWith from "lodash/unionWith";
+import { useApiCall } from "../Api";
+import { useApp } from "../App";
+import * as api from "../lib/api";
+import act from "../actions/methods";
+
+export const PublicProposalsContext = React.createContext();
+export const usePublicProposals = () =>
+  React.useContext(PublicProposalsContext);
+
+export const PublicProposalsProvider = ({ children }) => {
+  const { redux } = useApp();
+  const [voteStatus, isLoadingVoteStatus, voteStatusError] = useApiCall([
+    "proposalsVoteStatus"
+  ]);
+  const [
+    proposalsTokens,
+    isLoadingTokenInventory,
+    tokenInventoryError
+  ] = useApiCall(["tokenInventory"]);
+  const [isLoadingProposals, setLoadingProposals] = useState(false);
+  const [proposals, setProposals] = useState([]);
+  const [proposalsError, setProposalsError] = useState(false);
+
+  const onFetchVettedByTokens = useCallback(async tokens => {
+    try {
+      setLoadingProposals(true);
+      setProposalsError(false);
+      const fetched = await Promise.all(
+        tokens.map(t => api.proposal(t).then(r => r.proposal))
+      );
+
+      setProposals(proposals =>
+        unionWith(fetched, proposals, (a, b) => {
+          return a.censorshiprecord.token === b.censorshiprecord.token;
+        })
+      );
+    } catch (e) {
+      setProposalsError(e);
+    }
+    setLoadingProposals(false);
+  }, []);
+
+  useEffect(
+    function updateReduxWithVoteStatus() {
+      if (!voteStatus) return;
+      // This is an example of updating redux store with data
+      // For compatibility during refactors
+      redux.dispatch(
+        act.RECEIVE_PROPOSALS_VOTE_STATUS({
+          ...voteStatus,
+          success: true
+        })
+      );
+    },
+    [voteStatus]
+  );
+
+  useEffect(
+    function updateReduxWithVoteStatusError() {
+      if (!voteStatusError) return;
+      // This is an example of updating redux store with data
+      // For compatibility during refactors
+      redux.dispatch(act.RECEIVE_PROPOSALS_VOTE_STATUS(null, voteStatusError));
+    },
+    [voteStatusError]
+  );
+
+  const value = {
+    isLoading:
+      isLoadingProposals || isLoadingVoteStatus || isLoadingTokenInventory,
```

Comment:
The loading of proposals is handled in a different way in the inner (`components/PublicproposalsPage/PublicProposals.js`) component, so it should not trigger the global loading of the component. This loading management can, however, be pulled into the provider, if we want to.
```suggestion 
       isLoadingVoteStatus || isLoadingTokenInventory,
```

Reply:
> The components/PublicProposalsPage/* files would move to /PublicProposalsPage/ and more logic from the PubllicProposals component would get moved into provider if I continue with this. They are left as is to minimize the changes to highlight the more important bits.